stablehlo/tests/ops_stablehlo.mlir

```diff
@@ -4177,6 +4177,14 @@ func.func @get_dimension_size(%I: tensor<1x128x512xf32>) -> tensor<i32> {
 
 // -----
 
+func.func @get_dimension_size_invalid_dimension(%I: tensor<1x128x512xf32>) -> tensor<i32> {
+  // expected-error@+1 {{requires dimension attribute in range [0, 3); found (-1)}}
+  %size = "stablehlo.get_dimension_size"(%I) {dimension = -1 : i64} : (tensor<1x128x512xf32>) -> tensor<i32>
```

Comment:
Looks like `verifyDimAttr` accepts negative dimensions if the input is unranked. Let's fix that?